core/src/main/scala/codesearch/core/regex/space/SpaceInsensitive.scala

```diff
@@ -0,0 +1,37 @@
+package codesearch.core.regex.space
+
+import codesearch.core.regex.lexer.{StringAssembler, Tokenizer}
+import codesearch.core.regex.lexer.tokens.{SpecialSymbol, Token}
+
+object SpaceInsensitive {
+
+  /**
+    * Will make string insensitive to space
+    *
+    * @param query string query. Example: "hello world   kek ?"
+    * @return string with added symbols "+" for insensitive of spaces. Example: "hello +world   kek( +)?"
+    */
+  def spaceInsensitiveString(query: String): String = {
+    val tokens: Seq[Token] = Tokenizer.parseStringWithSpecialSymbols(query)
+
+    val selectMoreSpaces: List[Token] =
+      List(SpecialSymbol(")"), SpecialSymbol("+"), SpecialSymbol(" "), SpecialSymbol("("))
+
+    val addedRegexForSpaceInsensitive: Seq[Token] = tokens
+      .foldLeft(List.empty[Token]) {
+        case (result @ SpecialSymbol(" ") :: SpecialSymbol(" ") :: _, current) => current :: result
```

Comment:
You can simplify this by checking whether the first symbol is a space and _then_ matching on the second symbol.

Reply:
I mean something like this:

```scala
match result {
    case SpecialSymbol(" ") :: SpecialSymbol(" ") :: _ =>
        current :: result
    case SpecialSymbol(" ") :: _ => match current {
        SpecialSymbol("+") => current :: result
        SpecialSymbol("*") => current :: result
        SpecialSymbol("{") => current :: result
        SpecialSymbol("?") => current :: (selectMoreSpaces ::: result.tail)
        SpecialSymbol(" ") => current :: result
        _ => current :: SpecialSymbol("+") :: result
    }
    case _ =>
        current :: result
}
```